crypto/armcap.c

```diff
@@ -272,6 +291,9 @@ void OPENSSL_cpuid_setup(void)
         (OPENSSL_armcap_P & ARMV7_NEON)) {
             OPENSSL_armv8_rsa_neonized = 1;
     }
+    if ((MIDR_IS_CPU_MODEL(OPENSSL_arm_midr, ARM_CPU_IMP_ARM, ARM_CPU_PART_V1) || m1_flag)
+                            && (OPENSSL_armcap_P & ARMV8_SHA3))
+        OPENSSL_armv8_aes_unroll8_eor3 = 1;
```

Comment:
It is somehow odd, that this is not a bit in OPENSSL_armcap_P like every other feature here.
If this is a CPU-Feature, why is this not a bit like the other feature bits?

Reply:
OPENSSL_armcap_P depends on the environment setting, which tells us which features it supports before running.  Here we will detect the feature during runtime, so it's a bit different.

Reply:
Yes, but I think it shouldn't be handled different.
For instance when you look at the output of `openssl speed aes`
You should see the auto-detected OPENSSL_armcap bits,
but you can't see if `OPENSSL_armv8_aes_unroll8_eor3` is enabled or not.
Likewise for `OPENSSL_armv8_rsa_neonized`, but I really don't see why
these are not some bits in the OPENSSL_armcap, especially when you
look at the fragile feature detection logic of the `m1_flag`

Reply:
Emm. Firstly, it's odd to mix the features and micro-architectures together. Secondly, there are so many micro-architectures, and we are afraid that it's difficult to list all of them in the OPENSSL_armcap_P. So we just add run-time micro-architecture detection into Openssl to provide flexible ways to detect kinds micro-architectures of the CPU information.  About the seed test "show OPENSSL_armcap bits", we can talk about  that whether it also need to show the micro-architectures information.

Reply:
Hi all,  seems it's a long time after I reply this thread. So, what's your opinion?  Can someone kindly give us some suggestions? thanks.